client/src/modules/fundingSources/fundingSources.service.js

```diff
@@ -0,0 +1,29 @@
+angular.module('bhima.services')
+  .service('FundingSourceService', FundingSourceService);
+
+FundingSourceService.$inject = ['PrototypeApiService', 'util', '$uibModal'];
+
+/**
+ * @class FundingSourceService
+ * @extends PrototypeApiService
+ *
+ * @description
+ * Encapsulates common requests to the /funding_sources/ URL.
+ */
+function FundingSourceService(Api, util, $uibModal) {
+  const baseUrl = '/funding_sources/';
+  const tagKeys = ['uuid', 'label', 'code'];
+  const service = new Api(baseUrl);
+
+  service.clean = tag => util.maskObjectFromKeys(tag, tagKeys);
+
+  service.createUpdateFundingSourcesModal = (fundingSource) => {
+    $uibModal.open({
```

Comment:
You should call `return $uibModal.open({ ... }).instance` here, so that the consumers of this API will be notified when the modal has closed.